third-party/sysdeps/linux/libdrm/inline_amdgpu_ids.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+# Reads the data/amdgpu.ids file and replaces the amdgpu/amdgpu_asic_id.c file
+# with a fully inlined table that does not need to consult a fixed place on
+# the filesystem.
+
+SOURCE_DIR="${1:?Expected source directory}"
+
+generate() {
+  echo '
+#include <ctype.h>
+#include <stdint.h>
+#include <string.h>
+#include "amdgpu_drm.h"
+#include "amdgpu_internal.h"
+struct inline_amdgpu_id {
+  uint32_t did;
+  uint32_t rid;
+  const char *name;
+};
+'
+  # Output an inline table.
+  echo "static struct inline_amdgpu_id inline_amdgpu_ids[] = {"
+
+  # Syntax: device_id,  revision_id,  product_name
+  # Fields are separated by {comma} {tab}.
+  while read -r line; do
+    if [[ "$line" =~ ^# ]]; then
+      # Skip comment lines.
+      continue
+    fi
+    IFS=$',\t' read -ra parts <<< "$line"
+    if [[ ${#parts[@]} != 3 ]]; then
+      # Skip any lines that are not three fields. This also skips blanks and
+      # the file version leading line.
+      continue
+    fi
+    echo "  {0x${parts[0]}, 0x${parts[1]}, \"${parts[2]}\"},"
+  done < "${SOURCE_DIR}/data/amdgpu.ids"
+  echo "};"
+
+  # And export a lookup function.
+  echo "
+void amdgpu_parse_asic_ids(struct amdgpu_device *dev) {
+  size_t count = sizeof inline_amdgpu_ids / sizeof inline_amdgpu_ids[0];
```

Comment:
```suggestion
  const size_t count = sizeof inline_amdgpu_ids / sizeof inline_amdgpu_ids[0];
```
Should this be const?